jcommune-service/src/test/java/org/jtalks/jcommune/service/bb2htmlprocessors/UrlToLinkConvertPostProcessorTest.java

```diff
@@ -0,0 +1,105 @@
+                "деятельности играет важную роль в формировании дальнейших направлений развития."+
+                "\n\n" +
+                "Разнообразный и богатый опыт постоянное информационно-пропагандистское обеспечение нашей деятельности позволяет оценить значение модели развития. Задача организации, в особенности же реализация намеченных плановых заданий играет важную роль в формировании форм развития.";
+        assertEquals(postProcessor.postProcess(testText), testText);
+    }
+
+    @Test(dataProvider = "preProcessingCommonLinks")
+    void testUrlHighlighting(String incominText, String outcomingText) {
+        assertEquals(postProcessor.postProcess(incominText), outcomingText);
+    }
+
+    @Test(dataProvider = "preProcessingJoinedCommonLinks")
+    void testUrlHighlightingForJoinedCases(String incominText, String outcomingText) {
+        assertEquals(postProcessor.postProcess(incominText), outcomingText);
+    }
+
+    @DataProvider
+    public Object[][] preProcessingCommonLinks() {
+        return new Object[][] {
+                {"", ""},
+                {"http://javatalks.ru/common",
+                        "<a href=\"http://javatalks.ru/common\">http://javatalks.ru/common</a>"},
+                {"www.javatalks.ru/common",
+                        "<a href=\"http://www.javatalks.ru/common\">www.javatalks.ru/common</a>"},
+                {"ftp.javatalks.ru/common",
+                        "<a href=\"ftp://ftp.javatalks.ru/common\">ftp.javatalks.ru/common</a>"},
+                {"ftp://javatalks.ru/common",
+                        "<a href=\"ftp://javatalks.ru/common\">ftp://javatalks.ru/common</a>"},
+                {"<div class=divclass>http://javatalks.ru/common</div>",
+                        "<div class=divclass><a href=\"http://javatalks.ru/common\">http://javatalks.ru/common</a></div>"},
+                {"Text ftp.javatalks.ru/common text \n text http://javatalks.ru/common text and text \n the end.",
+                        "Text <a href=\"ftp://ftp.javatalks.ru/common\">ftp.javatalks.ru/common</a> text \n text <a href=\"http://javatalks.ru/common\">http://javatalks.ru/common</a> text and text \n the end."},
+                {"<pre class=cls>http://google.com</pre>",
+                        "<pre class=cls>http://google.com</pre>"},
+                {"<pre>http://google.com</pre>",
+                        "<pre>http://google.com</pre>"},
+                {"<pre class=cls>http://google.com\nwww.ya.ru\nabc abc abc ftp.server.com</pre>",
+                        "<pre class=cls>http://google.com\nwww.ya.ru\nabc abc abc ftp.server.com</pre>"},
+                {"Text <pre class=cls>http://google.com\nwww.ya.ru\nabc abc abc ftp.server.com</pre> www.ya.ru text",
+                        "Text <pre class=cls>http://google.com\nwww.ya.ru\nabc abc abc ftp.server.com</pre> <a href=\"http://www.ya.ru\">www.ya.ru</a> text"},
+                {"<a href=http://www.google.com>www.google.com</a>",
```

Comment:
This seems like a good case for a separate data provider. Let's split this provider into 2 - the one that expects the text to be modified and the one that expects that nothing changes. These should also be 2 different tests